src/vrcosclib/Utility/OscUtility.Connection.cs

```diff
@@ -48,29 +48,35 @@ public static int ReceivePort
 
     public static int SendPort
     {
-        get => _destination.Port;
+        get => Destination.Port;
         set
         {
-            //Do we need this setter? Doesn't look so good...
-            Destination = new IPEndPoint(Destination.Address, value);
+            if (_client != null)
+            {
+                _client.Destination.Port = value;
+            }
+            _destination.Port = value;
         }
     }
 
     public static string VrcIPAddress
     {
-        get => _destination.Address.ToString();
+        get => Destination.Address.ToString();
         set
         {
-            //Do we need this setter? Doesn't look so good...
-            //May throw
-            Destination = new IPEndPoint(IPAddress.Parse(value), Destination.Port);
+            var ip = IPAddress.Parse(value);
+            if (_client != null)
+            {
+                _client.Destination.Address = ip;
+            }
+            _destination.Address = ip;
         }
```

Comment:
Will this work in a real environment? 
Previously every time port or IP changes, the client will be recreated. The `IPEndPoint.Address` gives you an illusion that it will change the underlying socket's endpoint. I checked the `OscClient`, the socket is created at the .ctor. Changing `IPEndPoint.Address` won't do anything (Address property should be readonly...) My previous commit followed that behaviour, to recreate.  The `IPEndPoint` in OscClient will do nothing but as a parameter reference, changing the content of it won't do anything.

tbh I don't quite understand why `IPEndPoint` was made to be mutable, looks like a flaw in BCL.

Reply:
or just leave the Destination and remove the other two?

Reply:
I don't know that why you want `Destination`, but I think `Destination` is unnecessary in this library.
The `Destination` give us unexpected behaviors.

Reply:
The intention was to simplify the verification, such that a given `IPEndPoint` is always valid and make VrcIP readonly, then we have much less place to handle. But since `IPEndPoint` is sadly mutable, we may have misleading as you mentioned. Let's remove it for now. Some tests may need to improve as well. Since check mutated `Destination` in `OscClient` won't prove anything, should check if the client is recreated.